### PR TITLE
Add sigmoid to torch.nn.functional

### PR DESCRIPTION
**Pull request by kamathhrishi**
## Description
This PR adds sigmoid function to torch.nn.functional. The other approximations could also be added in the future when grad functions are made available. 

## How has this been tested?
- Added a unit test for the same

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by arturomf94 — APPROVED** [review]
LGTM 👍

**Review by gmuraru — APPROVED** [review]
LGTM!
